### ISSUE TITLE
5 minutes (download) + 5 minutes (booting up)

### ISSUE THREAD

**Issue by icy** [mondeja/solaris-vm-action#1]
I've looked at the recent action logs,, looked like

* it takes 5 minutes to download the image
* it takes another 5 minutes to boot the vm

I'd suggest to build own our vm instead :D make them small and reconfigure boot configuration. The default one from Sun may not be correctly /ideally configured ;)

**Comment by mondeja** [comment]
> I'd suggest to build own our vm instead :D make them small and reconfigure boot configuration.

I've made some changes in the machine before upload it, deleting most of the uneeded packages and disabling services. I still have to check if decreasing the disk size decreases final ova size. Also still have to try a recovering state strategy, but unsure if it is possible, or store in cache the uncompressed ova file... 🤔 

Do you have any sugestion about how to improve the machine or the script? You can download the ova file with:

```bash
curl -Z --parallel-max 8 "https://raw.githubusercontent.com/mondeja/solaris-vm-action-ova/master/ova/sol-11_4-part[00-75].zip" -o 'sol-11_4-part#1.zip'
cat sol-11_4-part* > sol-11_4.zip
unzip sol-11_4.zip
rm -f sol-11_4-part*.zip
```

**Comment by icy** [comment]
@mondeja I will have a look tomorrow. Basically you will need to create new vm from the scratch, because vbox image tend to be expanded (even _if_ you already removed something from it). They have two versions, one is for server, the other is for desktop. Are we using the server/base version?

edit: fix typo.

**Comment by mondeja** [comment]
Here is the basic TODO list for performance improvements:

- [x] Caching OVA file. Using `actions/checkout`, the OVA file can be cached and the ZIP file doesn't need to be downloaded in subsecuent runs:
   ```yaml
   - uses: actions/cache@v2
     with:
       key: sol-11_4
       path: |
         sol-11_4.ova
   ```
   > 5 minutes (download)

    With cache, subsequent runs tjust akes ~40s downloading the OVA file.

+ ~~Recover state from previous run starting the virtual machine (in development under tag [`dev`](https://github.com/mondeja/solaris-vm-action/releases/tag/dev)).~~ Not possible
+ ~~Install virtualbox guest additions? I've read somewhere that could increase performance.~~ Takes time and is not easy because the license agreement.
+ [ ] Disable more services in machine?
+ [ ] Smaller machine from scratch.

**Comment by mondeja** [comment]
Closed in favour of #2
